Source/WebKit/Shared/TextExtractionToStringConversion.cpp

```diff
@@ -400,100 +447,203 @@ static void addPartsForItem(const TextExtraction::Item& item, std::optional<Node
                 break;
             }
 
-            if (!containerString.isEmpty())
-                parts.append(WTFMove(containerString));
+            if (aggregator.useHTMLOutput()) {
+                String tagName;
+                if (containerType == TextExtraction::ContainerType::Root)
+                    tagName = "body"_s;
+                else if (!item.nodeName.isEmpty())
+                    tagName = item.nodeName.convertToASCIILowercase();
+
+                if (!tagName.isEmpty()) {
+                    auto attributes = partsForItem(item, aggregator);
+                    if (attributes.isEmpty())
+                        parts.append(makeString('<', tagName, '>'));
+                    else
+                        parts.append(makeString('<', tagName, ' ', makeStringByJoining(attributes, " "_s), '>'));
+                }
+            } else {
+                if (!containerString.isEmpty())
+                    parts.append(WTFMove(containerString));
 
-            parts.appendVector(partsForItem(item, aggregator));
+                parts.appendVector(partsForItem(item, aggregator));
+            }
             aggregator.addResult(line, WTFMove(parts));
         },
         [&](const TextExtraction::TextItemData& textData) {
             addPartsForText(textData, partsForItem(item, aggregator), WTFMove(enclosingNode), line, aggregator);
         },
         [&](const TextExtraction::ContentEditableData& editableData) {
-            parts.append("contentEditable"_s);
-            parts.appendVector(partsForItem(item, aggregator));
+            if (aggregator.useHTMLOutput()) {
+                auto attributes = partsForItem(item, aggregator);
+                if (attributes.isEmpty())
+                    parts.append(makeString('<', item.nodeName.convertToASCIILowercase(), '>'));
+                else
+                    parts.append(makeString('<', item.nodeName.convertToASCIILowercase(), ' ', makeStringByJoining(attributes, " "_s), '>'));
+
+                if (editableData.isPlainTextOnly)
+                    parts.append("contenteditable=plaintext-only"_s);
```

Comment:
should `plaintext-only` be quoted? Something like this?
```suggestion
                    parts.append("contenteditable='plaintext-only'"_s);
```

Reply:
Makes sense — that would be consistent with most of the other attributes.